src/sentry/models/event.py

```diff
@@ -264,8 +266,47 @@ def checksum(self):
         return ''
 
     def get_email_subject(self):
-        return '[%s] %s: %s' % (
-            self.project.get_full_name().encode('utf-8'),
-            six.text_type(self.get_tag('level')).upper().encode('utf-8'),
-            self.title.encode('utf-8')
+        template = self.project.get_option('mail:subject_template')
+        if template:
+            template = EventSubjectTemplate(template)
+        else:
+            template = DEFAULT_SUBJECT_TEMPLATE
+        return truncatechars(
+            template.safe_substitute(
+                EventSubjectTemplateData(self),
+            ).encode('utf-8'),
+            128,
         )
+
+
+class EventSubjectTemplate(string.Template):
+    idpattern = r'(tag:)?[_a-z][_a-z0-9]*'
+
+
+class EventSubjectTemplateData(object):
+    tag_aliases = {
+        'release': 'sentry:release',
```

Comment:
I feel like this must exist elsewhere, but I couldn't find it. Also, not sure if we should do anything special for `sentry:user` or others…?